docs/docs/components/guidelines/README.md

```diff
@@ -268,39 +288,51 @@ out](https://pipedream.com/community/c/dev/11).
 
 ##### Prop Definitions
 
-Whenever possible, reuse existing [prop definitions](/components/api/#prop-definitions-example).
+Whenever possible, reuse existing [prop
+definitions](/components/api/#prop-definitions-example).
 
 If a prop definition does not exist and you are adding an app-specific prop that
 may be reused in future components, add it as a prop definition to the app file.
 Prop definitions will also be surfaced for apps the Pipedream marketplace.
 
 ##### Methods
 
-Whenever possible, reuse
-[methods](/components/api/#methods)
-defined in the app file. If you need to use an API for which a method is not
-defined and it may be used in future components, define a new method in the app
-file.
+Whenever possible, reuse [methods](/components/api/#methods) defined in the app
+file or within a [common module](#common-files-optional).
 
-Use the [JS Docs](https://jsdoc.app/about-getting-started.html) pattern for
-lightweight documentation of each method in the app file. Provide a description
-and define @params and @returns block tags (with default values if applicable —
-e.g., `[foo=bar]`). This data will both help with reusability and will be
-surfaced in documentation for apps in the Pipedream marketplace. For example:
+As a rule of thumb, every interaction with the app-specific API should go
```

Comment:
Technically we don't call the `.app` files "components", so I would change to "app file" and link to `#app-files`